editor/views/rai.py

```diff
@@ -0,0 +1,249 @@
+            event=RaiEvent.RAI_DATA_BIAS,
+        )
+
+        one_to_many_property(
+            title="**Personal and sensitive information**",
+            metadata=metadata,
+            attributes=metadata.personal_sensitive_information,
+            key="metadata-personal-sensitive-information_",
+            label="Personal and sensitive information, if contained within the dataset, can play an important role in the mitigation of any risks and the responsible use of the datasets",
+            event=RaiEvent.RAI_SENSITIVE,
+        )
+
+        key = "metadata-social-impact"
+        st.text_area(
+            label=("**Social impact**. Discussion of social impact, if applicable"),
+            key=key,
+            value=metadata.data_social_impact,
+            on_change=handle_rai_change,
+            args=(RaiEvent.RAI_DATA_SOCIAL_IMPACT, metadata, key),
+        )
+
+        one_to_many_property(
+            "**Data limitations**",
+            metadata,
+            metadata.data_limitations,
+            "metadata-data-limitations_",
+            "Known limitations - Data generalization limits (e.g related to data distribution, data quality issues, or data sources) and on-recommended uses.",
+            RaiEvent.RAI_DATA_LIMITATION,
+        )
+
+        key = "metadata-data-maintenance"
+        st.text_area(
+            label=(
+                "**Data release maintenance**. Versioning information in terms of the updating timeframe, the maintainers, and the deprecation policies. "
+            ),
+            key=key,
+            value=metadata.data_release_maintenance_plan,
+            on_change=handle_rai_change,
+            args=(RaiEvent.RAI_MAINTENANCE, metadata, key),
+        )
+
+
+def one_to_many_property(
+    title: str, metadata: Metadata, attributes, key: str, label: str, event: str
+):
+    """Generates a one to many cardinality property. Attributes should be empty, have one element or being a list of elements"""
+    with st.expander(title, expanded=True):
+        if attributes:
+            if not isinstance(attributes, list):
+                attributes = [attributes]
+            for index, single_attribute in enumerate(attributes):
+                key = key + str(index)
+                st.text_area(
+                    label=(label),
+                    key=key,
+                    value=single_attribute,
+                    on_change=handle_rai_change,
+                    args=(event, metadata, key, index),
+                )
+        else:
+            key = key + "0"
+            st.text_area(
+                label=(label),
+                key=key,
+                on_change=handle_rai_change,
+                args=(event, metadata, key),
+            )
+        add, remove = st.columns(2)
+        with add:
+            if st.button("+ add", key=key + "add"):
+                if attributes:
+                    attributes.append("")
+                    st.rerun()
```

Comment:
Prefer callbacks rather than st.rerun (see https://docs.streamlit.io/develop/api-reference/execution-flow/st.rerun). Maybe by having an `on_add`/`on_remove`? That'd allow to distinguish between the case list/not list.

```python
one_to_many_property(
  attributes=...,
  handle_add=...,
  handle_remove=...,
)


if handle_add:
  st.button("+ add", key=f"add_{key}", on_click=handle_add)
```